app/views/shared/_ingredient_form.html.erb

```diff
@@ -0,0 +1,30 @@
+<%= form_with(model: recipe_food) do |form| %>
+  <% if recipe_food.errors.any? %>
+    <div style="color: red">
+      <h2><%= pluralize(recipe_food.errors.count, "error") %> prohibited this recipe_food from being saved:</h2>
+
+      <ul>
+        <% recipe_food.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <%= form.hidden_field :recipe_id, value: @recipe.id %>
+  <%= form.label :quantity, style: "display: block" %>
+  <%= form.text_field :quantity %> <br>
+  <%= form.label :food, class: 'form-label' %>
+  <%= form.text_field :food, class: 'form-control', list: "datalistOptions", placeholder: "Type to search owned ingredients..." %>
+  <datalist id="datalistOptions">
+  <% if current_user.present? %>
+    <% current_user.foods.each do |food| %>
+      <option value="<%= food.name %>">
+  <% end %>
```

Comment:
- [ ] Good job so far guys 👏 When I tried to add ingredients(food) for **the recipe which is not in the food items already in the system, It gives me an error because it can't find it of course.** However, the requirements are a bit different here. Someone should be able to add any food that he/she wants as an ingredient to the recipe. Then it is the **purpose of the `Shopping list` that we have there to list the foods that are listed in recipes while they are not in our inventory(foods already in the system)**  Please refactor this logic you used here so as to accept and record any food  item 😃 👍 
![image](https://user-images.githubusercontent.com/90222110/221292074-4cd4691a-3270-48e6-a075-959017695ecf.png)

Reply:
@BertrandConxy do you have any tips about that? Because recipes do not have ingredients themselves